packages/eslint-plugin-react-hooks/__tests__/ESLintRulesOfHooks-test.js

```diff
@@ -479,6 +479,32 @@ const tests = {
         }
       `,
     },
+    {
+      // Nested component declarations are bad but should be covered by other ESLint rules
+      code: normalizeIndent`
+        function App() {
+          return createElement(Child, {
+            Component: () => {
```

Comment:
This is already considered a component in the latest release. Just not when it's written as JSX